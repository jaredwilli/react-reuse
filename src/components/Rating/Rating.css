.reviews-rating {
    display: flex;
}

.reviews-rating .star-rating a {
  display: inline-block;
}

.reviews-rating .star-average {
    border-right: 1px solid #e8e8e8;
    margin: 0 10px;
    padding-right: 10px;
    display: flex;
    align-items: center;
    justify-content: center;
}

.reviews-rating .star-total {
    display: flex;
    align-items: center;
    justify-content: center;
}
