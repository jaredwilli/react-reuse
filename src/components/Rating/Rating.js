import React from 'react';
import PropTypes from 'prop-types';

import StarIcon from '../StarIcon';

import './Rating.css';

/** Star Rating component that sets the selection on click and has hover animation */
class Rating extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            rating: this.props.rating || null,
            tempRating: null
        };
    }

    onClick = rating => {
        if (this.props.rating) { return; } // disable

        this.setState({
            tempRating: rating,
            rating
        });
    };

    onMouseOver = rating => {
        if (this.props.rating) { return; } // disable

        this.setState({
            tempRating: this.state.rating,
            rating
        });
    };

    onMouseOut = () => {
        if (this.props.rating) { return; } // disable

        this.setState({
            rating: this.state.tempRating
        });
    };

    render = () => {
        const fill = '#29292A';
        const stars = [];

        for (let i = 1; i <= this.props.scale; i++) {
            const isHalf = this.state.rating % 1 > 0;
            let fillType = 'transparent';

            if (this.state.rating !== null) {
                if (this.state.rating >= i) {
                    fillType = 'full';
                } else if (isHalf && Math.ceil(this.state.rating) === i) {
                    fillType = 'half';
                }
            }

            stars.push(
                <a
                    key={i}
                    className="star-link"
                    style={(!this.props.rating) ? { cursor: 'pointer' } : {}}
                    onClick={() => this.onClick(i)}
                    onMouseOver={() => this.onMouseOver(i)}
                    onMouseOut={() => this.onMouseOut()}
                >

                    <StarIcon fillType={fillType} fill={fill} size={this.props.size} />
                </a>
            );
        }

        return (
            <div className="reviews-rating">
                <span className="star-rating">
                    {stars}
                </span>

                {this.props.showAverage && <span className="star-average">{this.props.rating}</span>}
                {this.props.totalRatings && <span className="star-total">{this.props.totalRatings} reviews</span>}
            </div>
        );
    };
}

Rating.propTypes = {
    /** Show the average rating value or not */
    showAverage: PropTypes.bool,

    /** The total number of reviews that have included a rating */
    totalRatings: PropTypes.number,

    /** The size of the stars in the rating component */
    size: PropTypes.number,

    /** The rating scale number for how many stars */
    scale: PropTypes.number,

    /** The rating value to set initially if needed */
    rating: PropTypes.number
};

Rating.defaultProps = {
    totalRatings: null,
    rating: null,
    size: 32,
    scale: 5
};

export default Rating;
