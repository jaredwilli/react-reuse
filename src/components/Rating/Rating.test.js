import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Rating from './Rating';

describe('Rating', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<Rating />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with a default rating value', () => {
        const tree = renderer.create(<Rating rating={2} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with a default rating half value', () => {
        const tree = renderer.create(<Rating rating={0.5} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with a default rating half value and size prop', () => {
        const tree = renderer.create(<Rating rating={0.5} size={20} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
