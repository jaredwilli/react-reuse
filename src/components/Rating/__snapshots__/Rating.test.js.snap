// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders the component correctly 1`] = `
<div
  className="reviews-rating"
>
  <span
    className="star-rating"
  >
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
  </span>
</div>
`;

exports[`Rating renders with a default rating half value 1`] = `
<div
  className="reviews-rating"
>
  <span
    className="star-rating"
  >
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="#29292A"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
  </span>
</div>
`;

exports[`Rating renders with a default rating half value and size prop 1`] = `
<div
  className="reviews-rating"
>
  <span
    className="star-rating"
  >
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={20}
        id="star-icon"
        viewBox="0 0 32 32"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="#29292A"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={20}
        id="star-icon"
        viewBox="0 0 32 32"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={20}
        id="star-icon"
        viewBox="0 0 32 32"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={20}
        id="star-icon"
        viewBox="0 0 32 32"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={20}
        id="star-icon"
        viewBox="0 0 32 32"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
  </span>
</div>
`;

exports[`Rating renders with a default rating value 1`] = `
<div
  className="reviews-rating"
>
  <span
    className="star-rating"
  >
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="#29292A"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="#29292A"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
    <a
      className="star-link"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={Object {}}
    >
      <svg
        className="star-icon"
        height={32}
        id="star-icon"
        viewBox="0 0 32 32"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="full"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36,4,9.36,9,.12-6.5,6.36,1.53,9Z"
          fill="transparent"
          stroke="#29292A"
          transform="translate(-1.04 -0.23)"
        />
        <path
          className="half"
          d="M15.25,22.08l-8,4.24,1.53-9L2.25,11l9-.12,4-9.36Z"
          fill="transparent"
          stroke="transparent"
          transform="translate(-1.04 -0.23)"
        />
      </svg>
    </a>
  </span>
</div>
`;
